lint/returnAfterLogFatal_checker.go

```diff
@@ -0,0 +1,46 @@
+package lint
+
+//! Detects return statements that follow panic/fatal logging.
+//
+// @Before:
+// log.Fatal("exits function")
+// return
+//
+// @After:
+// log.Fatal("exits function")
+
+import (
+	"go/ast"
+)
+
+func init() {
+	addChecker(&returnAfterLogFatalChecker{}, attrExperimental)
+}
+
+type returnAfterLogFatalChecker struct {
```

Comment:
What about calling it `unreachableReturn` or `deadReturn`?
We may add more common functions later, like `os.Exit` to the list, for example.
(Note that I can do renaming after the merge myself, but it's better to decide a name together.)